### PR TITLE
Changes POST example in Basic.md to send parameters as JSON

### PR DESCRIPTION
**Pull request by pedrovereza**
The POST example in `Basic.md` currently sends the parameters encoded in the URL. I don't think this is the most common way of sending parameters in a POST request 🤔  

I believe the usual case is to send parameters in the request body as JSON. Having this in the most basic example may be more helpful to people using Moya for the first time.

### CODE REVIEW

**Comment by vzsg** [comment]
Why not both?

**Comment by pedrovereza** [comment]
@vzsg Good question! 

What are the most basic (and common) cases that we should cover in this example? We don't have a GET with URL query parameters as well 🤔

**Comment by sunshinejr** [comment]
This example you've added is good, but we could also add the basic `URLEncoding` below/above the example with `JSONEncoding`, imho.

**Comment by pedrovereza** [comment]
@sunshinejr You mean adding just a note on how the POST request would look like if `URLEncoding` was used instead?

**Comment by sunshinejr** [comment]
I was thinking about doing one more request with different type of encoding, maybe just below the one with user, and then make a comment how would it look like. But your suggestion works as well, make a comment how could it look like with `URLEncoding`.

**Comment by pedrovereza** [comment]
@sunshinejr Doing another request feels more complete 👍 

Added a new endpoint to update a user passing the parameters in the URL, let me know what you think (cc. @vzsg)

File: docs/Examples/Basic.md [hunk 1]
```diff
@@ -25,32 +26,38 @@ extension MyService: TargetType {
...
-        case .createUser(let firstName, let lastName):
+        case .createUser(let firstName, let lastName), .updateUser(_, let firstName, let lastName):
             return ["first_name": firstName, "last_name": lastName]
         }
     }
+
     var parameterEncoding: ParameterEncoding {
-        return URLEncoding.default
+        switch self {
+        case .zen, .showUser, .showAccounts, .updateUser(_, _, _):
```
Comment (sunshinejr) [inline docs/Examples/Basic.md]: This is a minor one, but I think we can strip these `(_, _, _)` when we don't need any of the parameters in there.

Comment (BasThomas) [inline docs/Examples/Basic.md]: You can :)

Comment (pedrovereza) [inline docs/Examples/Basic.md]: Yep! Good catch

**Comment by pedrovereza** [comment]
Should be good now 😄

**Pull request merged**